Tasks/ExtractFilesV1/extractfilestask.ts

```diff
@@ -20,18 +21,32 @@ tl.debug('win: ' + win);
 var xpTarLocation: string;
 var xpUnzipLocation: string;
 // 7zip
-var xpSevenZipLocation: string;
-var winSevenZipLocation: string = path.join(__dirname, '7zip/7z.exe');
+let sevenZipLocation: string;
+let defaultWinSevenZipLocation: string = path.join(__dirname, '7zip/7z.exe');
 
 function getSevenZipLocation(): string {
+    if (customPathToSevenZipTool) {
+        tl.debug('Get 7z tool from user defined location');
+        return customPathToSevenZipTool;
+    }
+    
     if (win) {
-        return winSevenZipLocation;
-    } else {
-        if (typeof xpSevenZipLocation == "undefined") {
-            xpSevenZipLocation = tl.which('7z', true);
+        if (!sevenZipLocation) {
```

Comment:
Is this check necessary? The `sevenZipLocation` variable should always be `undefined` at that moment. I also believe that it's not necessary to define the variable outside the method's scope since there's no need to reuse it.

Reply:
There is one case when _getSevenZipLocation_ function is calling twice during task execution. If working with compressed tar archives seven zip tool will be called twice [here](https://github.com/microsoft/azure-pipelines-tasks/blob/eedecd000d67418dea8492247e7cbb3b17a0102f/Tasks/ExtractFilesV1/extractfilestask.ts#L295) and [here](https://github.com/microsoft/azure-pipelines-tasks/blob/eedecd000d67418dea8492247e7cbb3b17a0102f/Tasks/ExtractFilesV1/extractfilestask.ts#L300). I'm not sure that we need to run logic to find 7z on agent twice, since we can just reuse the result of previous function running. And also because of it _sevenZipLocation_ is not always has undefined value.

Reply:
@EzzhevNikita Thank you! Got it.